src/ol/events/SnapEvent.js

```diff
@@ -0,0 +1,11 @@
+import BaseEvent from './Event.js';
+
+const SNAP = 'snap';
+export class SnapEvent extends BaseEvent {
+  constructor(options) {
+    super(SNAP);
+    this.vertex = options.vertex;
+    this.vertexPixel = options.vertexPixel;
+    this.feature = options.feature;
```

Comment:
Had to follow existing pattern with enum SnapEventType , otherwise api-doc won't include SnapEvent properly.

![image](https://github.com/openlayers/openlayers/assets/104476360/326a0f10-e857-4600-96ce-9ff9a01ead4d)

![image](https://github.com/openlayers/openlayers/assets/104476360/fd695161-fe6a-4b32-a114-2a0172e5244e)